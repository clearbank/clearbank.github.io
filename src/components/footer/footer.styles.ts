import styled from 'styled-components'
import { Box } from 'rebass'

import { colors } from 'src/components/theme'

export const Container = styled(Box).attrs(() => ({ mt: '140px' }))``

export const MenuContainer = styled.div`
  display: flex;
  justify-content: space-between;
  margin: 0 7%;
  margin-left: 50px;
  padding: 0 5%;
  padding-left: 0;
  border-bottom: 1px solid ${colors.mischkaapprox};
`

export const ColumnContainer = styled.div`
  max-width: 200px;
`

export const Title = styled.h2`
  cursor: default;
  margin: 0;
  text-decoration: none;
  font-weight: 600;
  font-size: 20px;
  color: ${colors.brandDark}
`;

export const SubTitle = styled.a`
  line-height: 20px;
  text-decoration: none;
  cursor: pointer;
  font-weight: 400;
  font-size: 18px;
  color: ${colors.brandDark}
`

export const List = styled.ul`
  margin-top: 40px;
  list-style: none;
  padding: 0;
`

export const ListItem = styled.li`
  margin-bottom: 20px;
`

export const LogoWrapper = styled.div`
  max-width: 170px;
  display: block;
  margin: 40px 0px 20px 40px; 
`

export const CopywriteWrapper = styled.div`
  text-align: center;
  margin-bottom: 40px;
`

export const Copywrite = styled.p`
  max-width: 70%;
  margin-left: 50px;
  margin-right: auto;
  text-align: left;
  color: ${colors.brandSecondaryDarker};
`
