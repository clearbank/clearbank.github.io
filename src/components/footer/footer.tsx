import React from 'react'
import kebabCase from 'lodash.kebabcase'

import Logo from 'src/components/logo'

import * as Styles from './footer.styles'

import * as Types from 'src/components/header-menu/header-meny.types'

const Footer: React.FC<Types.IHeaderMenuProps> = ({ items }) => (
  <Styles.Container role='contentinfo'>
    <Styles.MenuContainer>
      {items.map(item => (
        <Styles.ColumnContainer key={item.menuItem.title}>
          <Styles.Title>
            {item.menuItem.title}
          </Styles.Title>
          <Styles.List>
            {item.subMenuItems.map(subMenuItem => (
              <Styles.ListItem key={subMenuItem.node.fields.slug}>
                <Styles.SubTitle
                  href={
                    `${item.menuItem.slug}#${kebabCase(subMenuItem.node.fields.title.toLowerCase())}`
                  }
                >
                  {subMenuItem.node.fields.title}
                </Styles.SubTitle>
              </Styles.ListItem>  
            ))}
          </Styles.List>
        </Styles.ColumnContainer>
      ))}
    </Styles.MenuContainer> 
    <Styles.CopywriteWrapper>
      <Styles.LogoWrapper>
        <Logo inverted />
      </Styles.LogoWrapper>

      <Styles.Copywrite className='long-primer'>
        Copyright ClearBank Ltd. Authorised by the Prudential Regulation Authority and regulated by the Financial Conduct Authority and the Prudential Regulation Authority
        (Financial Services Register number: 754568). Registered Address: ClearBank, 4th Floor, Prologue Works, 25 Marsh St, Bristol BS1 4AX
      </Styles.Copywrite>
    </Styles.CopywriteWrapper>
  </Styles.Container>
)

export default Footer
