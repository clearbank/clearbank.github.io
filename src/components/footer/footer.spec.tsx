import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import 'jest-styled-components'

import Component from './footer'

import * as Types from './footer.types'

let component: ReturnType<typeof render> = null
let fragment: DocumentFragment = null
let root: HTMLElement = null

const defaultProps: Types.FooterProps = {
  items: [],
}

beforeEach(() => {
  component = render(<Component {...defaultProps} children='children' />)

  fragment = component.asFragment()
  root = component.container.firstChild
})

afterEach(() => {
  fragment = null
  root = null
})

describe('Component', () => {
  test('matches snapshot', () => {
    expect(root).toMatchSnapshot()
  })

  test('has child elements', () => {
    expect(root.hasChildNodes()).toBeTruthy()
  })

  test('has offical company name in text', () => {
    expect(root.textContent.includes('Copyright ClearBank Limited')).toBeTruthy()
  })

  test('is not empty', () => {
    expect(fragment.textContent.length).toBeTruthy()
  })
})
