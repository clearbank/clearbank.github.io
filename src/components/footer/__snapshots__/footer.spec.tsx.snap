// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component matches snapshot 1`] = `
.c5 {
  width: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 20px;
  border-top: dashed 1px #E0E0E0;
}

.c4 {
  max-width: 170px;
  display: block;
  margin: auto;
  margin-bottom: 30px;
  margin-top: 30px;
}

.c3 {
  text-align: center;
  margin-bottom: 40px;
}

.c1 {
  padding: 0 20px;
}

.c2 {
  display: block;
  width: 100%;
  color: #6C7F98;
  font-size: 14px;
}

.c6 {
  white-space: nowrap;
  color: #6C7F98;
}

@media screen and (min-width:992px) {
  .c0 {
    -webkit-align-items: start;
    -webkit-box-align: start;
    -ms-flex-align: start;
    align-items: start;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

@media screen and (min-width:992px) {
  .c0 {
    padding: 30px;
  }
}

@media screen and (min-width:992px) {
  .c4 {
    margin-top: 0;
  }
}

@media screen and (min-width:768px) {
  .c3 {
    margin-top: 30px;
  }
}

@media screen and (min-width:992px) {
  .c3 {
    padding-right: 50px;
  }
}

@media screen and (min-width:992px) {
  .c2 {
    padding-right: 55px;
  }

  .c2:first-child {
    margin-top: 30px;
  }
}

<div
  class="c0 css-vurnku"
  role="contentinfo"
>
  <div
    class="c1"
  >
    <p
      class="c2"
    >
      Copyright ClearBank Ltd. Authorised by the Prudential Regulation Authority and regulated by the Financial Conduct Authority and the Prudential Regulation Authority (Financial Services Register number: 754568). Registered Address: ClearBank, 4th Floor, Prologue Works, 25 Marsh St, Bristol BS1 4AX.
    </p>
    <p
      class="c2"
    >
      ClearBank® provides customers with a Definitive Services Agreement, which covers the provision of indirect access services. Code of Conduct for Indirect Access Providers
    </p>
  </div>
  <div
    class="c3"
  >
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <img
          alt="ClearBank"
          src=""
        />
      </div>
    </div>
    <span
      class="c6 long-primer"
    >
      © Copyright 
      2023
       ClearBank
      <sup>
        ®
      </sup>
    </span>
  </div>
</div>
`;
